CORE-5005 Virtual node upgrade related entity changes

Tom-Fitzpatrick opened the pull request:
This PR updates VirtualNodeEntity with the fields needed to support CPI upgrade, as described in the [CPI upgrade design documentation](https://github.com/corda/platform-eng-design/blob/master/core/corda-5/corda-5.0/virtual-node/cpi-upgrade/database-schema-changes.md). The corresponding corda-api changes are being made in https://github.com/corda/corda-api/pull/803

The fields being added to VirtualNodeEntity are:
- flowP2pOperationalStatus
- flowStartOperationalStatus
- flowOperationalStatus
- vaultDbOperationalStatus
- operationInProgress
- vaultDDLConnectionId
- vaultDMLConnectionId
- cryptoDDLConnectionId
- cryptoDMLConnectionId
- uniquenessDDLConnectionId
- uniquenessDMLConnectionId

Additionally, this PR moves the connectionId fields from the HoldingIdentityEntity to VirtualNodeEntity.

Tom-Fitzpatrick committed:
add OperationalStatus to VirtualNodeEntity, move ConnectionIds from HoldingIdentity to VirtualNodeEntity

Tom-Fitzpatrick committed:
fix imports

Tom-Fitzpatrick committed:
reorder parameters for VirtualNodeEntity, tidy up uses

Tom-Fitzpatrick committed:
fix VirtualNodeWriterProcessorTests issue

Tom-Fitzpatrick committed:
update comments, fix failing test

Tom-Fitzpatrick committed:
fix detekt issues, unnecessary checks

sprintengine commented on libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/entities/VirtualNodeOperationEntity.kt:
```diff
@@ -0,0 +1,61 @@
+@Entity
+@Table(name = "virtual_node_operation", schema = CONFIG)
+@Suppress("LongParameterList")
+internal class VirtualNodeOperationEntity(
```
I think you also need to add this entity to `VirtualNodeEntities`

Tom-Fitzpatrick replied:
Done, thanks!

Tom-Fitzpatrick committed:
add @Id field to PK on VirtualNodeEntity

sprintengine commented on libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/entities/VirtualNodeEntity.kt:
```diff
@@ -3,21 +3,22 @@ package net.corda.libs.virtualnode.datamodel.entities
 import java.io.Serializable
 import java.time.Instant
-import java.util.Objects
+import java.util.*
```
I think there is an intellij configuration to get rid of * imports, maybe @vkolomeyko knows what this is / how to get this setting?

Tom-Fitzpatrick replied:
I had a look there and I think I found it in Settings -> Editor -> Code Style -> Kotlin -> Imports. There was a specific exception to always make java.util use "*". Hopefully that's it anyway.

Tom-Fitzpatrick committed:
update VIRTUAL_NODE_DB_TABLE import

Signed-off-by: TomFitzpatrick <tom.fitzpatrick@r3.com>

Tom-Fitzpatrick committed:
fix VirtualNodeEntity id

Tom-Fitzpatrick committed:
fix VirtualNodeEntitiesIntegrationTest

Tom-Fitzpatrick committed:
changing operationInProgress cascade

Tom-Fitzpatrick committed:
use random id for HoldingIdentity

Tom-Fitzpatrick committed:
fix conflict

Tom-Fitzpatrick committed:
VirtualNodeStateChangeResponse and VirtualNodeCreateResponse return vnode OperationalStatus

VirtualNodeStateChangeResponse and VirtualNodeCreateResponse return vnode OperationalStatus'

fix VirtualNodeWriterProcessorTests

remove unused import

Tom-Fitzpatrick committed:
fix set virtual node state test

Tom-Fitzpatrick committed:
fix VirtualNodeRpcTest

sprintengine commented on gradle.properties:
```diff
@@ -40,7 +40,7 @@ bouncycastleVersion=1.72
 # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
 #cordaApiVersion=5.0.0.xxx-SNAPSHOT
-cordaApiVersion=5.0.0.621-beta+
+cordaApiVersion=5.0.0.622-alpha-1674486979799
```
We will bump this to 622-beta+ after we merge the [corda-api](https://github.com/corda/corda-api/pull/803) change

Tom-Fitzpatrick committed:
addign prints to find error

Tom-Fitzpatrick committed:
trying to work out why findall() test is failing

Tom-Fitzpatrick committed:
use VNodeTestUtils.newVNode in VirtualNodeEntitiesIntegrationTest

Tom-Fitzpatrick committed:
update cordaApiVersion

Tom-Fitzpatrick committed:
fix cordaApiVersion

Tom-Fitzpatrick committed:
remove temporary changes

Tom-Fitzpatrick committed:
remove unnecessary import

sprintengine commented on libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepositoryImpl.kt:
```diff
@@ -55,37 +55,70 @@ class VirtualNodeRepositoryImpl : VirtualNodeRepository {
             val latestVirtualNodeInstance = findEntity(entityManager, holdingIdentityShortHash)
                 ?: throw VirtualNodeNotFoundException(holdingIdentityShortHash)
+
+            var operationalStatus = OperationalStatus.ACTIVE
```
Just for reviewers, for now we're converting the user's change state to the operational statuses here, in the repository.

As node upgrade evolves, we might need to move this elsewhere, its more of a business logic concern rather than the concern of a CRUD repository, but for now the logic here is OK.

One note is that we've changed the API here, before we expected "IN_MAINTENANCE" now we're expecting "MAINTENANCE", @vkolomeyko is this something we need to communicate to external tests / QA workflows? Alternatively we can change this to `if (newState == "in_maintenance" || newState == "maintenance")` (not sure if we already perform undercasing of the newState here, coud be worth calling `newState.toLowerCase()` - but preferably not as part of this PR because I'm keen to get these API changes merged.

vkolomeyko replied:
I think it will not harm to keep QA informed, but change of state shoudl not have had any material impact on the test results so far.

Tom-Fitzpatrick committed:
convert uuid to string

vkolomeyko reviewed (dismissed):
LGTM

vkolomeyko commented on gradle.properties:
```diff
@@ -40,7 +40,7 @@ bouncycastleVersion=1.72
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
 #cordaApiVersion=5.0.0.xxx-SNAPSHOT
-cordaApiVersion=5.0.0.623-beta+
```
Gentle reminder to update to `beta` version once it is available.

Tom-Fitzpatrick replied:
Done, thanks

Tom-Fitzpatrick committed:
update cordaApiVersion to beta

Tom-Fitzpatrick committed:
remove version number for +

vkolomeyko reviewed (approved):
LGTM
